hash_practice/exercises.py

```diff
@@ -1,20 +1,70 @@
 
+from audioop import reverse
+from re import L
+from tokenize import group
+
+
 def grouped_anagrams(strings):
     """ This method will return an array of arrays.
         Each subarray will have strings which are anagrams of each other
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n)
+        Space Complexity: O(n)
     """
-    pass
+
+    grouped_anagrams = {}
+
+    for word in strings:
+        unique_word = "".join(sorted(word))
+        if unique_word in grouped_anagrams:
+            grouped_anagrams[unique_word].append(word)
+        else:
+            grouped_anagrams[unique_word] = [word]
+    
+    output = []
+
+    for values in grouped_anagrams.values():
+        output.append(values)
+    
+    return output
+
 
 def top_k_frequent_elements(nums, k):
     """ This method will return the k most common elements
         In the case of a tie it will select the first occuring element.
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n)
+        Space Complexity: O(n)
     """
-    pass
+    if len(nums) == 0:
+        return []
+
+    num_map = {}
+
+    for i in nums:
+        if i in num_map:
+            num_map[i] += 1
+        else:
+            num_map[i] = 1
+    
+    if len(num_map) == k:
+        return nums
 
+    reverse_num_map = {}
+
+    for key in num_map:
+        value = num_map[key]
+        if value in reverse_num_map:
+            reverse_num_map[value - .5] = key
+        else:
+            reverse_num_map[value] = key

```

Comment:
This is going to fail if there are more than two numbers that have the same counts. For example, given [1,1,1,2,2,2,3,3,3,4,4], the 3 key in`reverse_num_map` will first get set to 2 and then get set to 3 (assuming the keys of `num_map` are processed as 1, 2, and 3 in that order, which isn't guaranteed).

So if k was 3 in this case, we'd incorrectly return 1 and 3 only.

This is minor enough that I do not feel it warrants moving down to Red, but it _is_ an issue so I'm probably going to have to mark it Yellow.